src/main/java/com/wojciechbarwinski/demo/epic_board_games_shop/legendaryWarehouse/LegendaryWarehouseAdapter.java

```diff
@@ -3,17 +3,36 @@
 import com.wojciechbarwinski.demo.epic_board_games_shop.clients.LegendaryWarehouseClient;
 import com.wojciechbarwinski.demo.epic_board_games_shop.dtos.OrderDataFromWarehouseDTO;
 import com.wojciechbarwinski.demo.epic_board_games_shop.dtos.OrderDataToWarehouseDTO;
+import com.wojciechbarwinski.demo.epic_board_games_shop.entities.Order;
+import com.wojciechbarwinski.demo.epic_board_games_shop.mappers.MapperFacade;
+import com.wojciechbarwinski.demo.epic_board_games_shop.repositories.OrderRepository;
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Component;
 
+@Slf4j
 @RequiredArgsConstructor
 @Component
 public class LegendaryWarehouseAdapter implements LegendaryWarehousePort {
 
     private final LegendaryWarehouseClient client;
+    private final MapperFacade mapper;
+    private final OrderRepository orderRepository;
+
 
     @Override
-    public OrderDataFromWarehouseDTO sendOrderToWarehouse(OrderDataToWarehouseDTO orderDataToWarehouseDTO) {
-        return client.sendOrderToWarehouse(orderDataToWarehouseDTO);
+    public void sendOrderToWarehouse(Order order) {
+        OrderDataToWarehouseDTO orderDataToWarehouseDTO = mapper.mapOrderToOrderDataToWarehouse(order);
+        log.info("Order with id {} is being sent to the warehouse...", order.getId());
+
+        try {
+            OrderDataFromWarehouseDTO orderDataFromWarehouseDTO = client.sendOrderToWarehouse(orderDataToWarehouseDTO);
+            log.info("Order with id {} was sent to the warehouse successfully", order.getId());
+            order.setOrderStatus(orderDataFromWarehouseDTO.status());
```

Comment:
Mam trochę wątpliwości co do tej linijki - ale to będzie lepiej przegadać na słuchawce